inc/azure_c_util/interlocked_if.h

```diff
@@ -0,0 +1,156 @@
+#ifndef INTERLOCKEDIFIF_H
+#define INTERLOCKEDIFIF_H
+
+#include <stdint.h>
+
+long InterlockedIFAdd( 
+    long volatile *Addend,
+    long           value
+);
+
+long InterlockedIFAnd( 
+    long volatile *Destination,
+    long          Value
+);
+
+short InterlockedIFAnd16(
+  short volatile *Destination,
+  short          Value
+);
+
+int64_t InterlockedIFAnd64(
+  int64_t volatile *Destination,
+  int64_t          Value
+);
+
+char InterlockedIFAnd8(
+  char volatile *Destination,
+  char          Value
+);
+
+long InterlockedIFCompareExchange(
```

Comment:
up to this point, the function names followed strictily the list on the right of this page. [https://docs.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedand8](https://docs.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedand8)

What happened to InterlockedCompareExchange128 ?:)

Reply:
Dan said we won't need the 128 bit function so I removed it.